core/src/main/java/com/linecorp/armeria/client/RestClientPreparation.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+
+import org.reactivestreams.Publisher;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.errorprone.annotations.FormatMethod;
+
+import com.linecorp.armeria.common.Cookie;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.JacksonObjectMapperProvider;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.ResponseEntity;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+import io.netty.util.AttributeKey;
+
+/**
+ * Prepares and executes a new {@link HttpRequest} for {@link RestClient}.
+ */
+@UnstableApi
+public final class RestClientPreparation implements RequestPreparationSetters {
+
+    private final WebClientRequestPreparation delegate;
+
+    RestClientPreparation(WebClient client, HttpMethod method, String path) {
+        delegate = client.prepare();
+        delegate.method(method);
+        delegate.path(path);
+    }
+
+    /**
+     * Sends the HTTP request and converts the JSON response body as the {@code T} object using the default
+     * {@link ObjectMapper}.
+     *
+     * @see JacksonObjectMapperProvider
+     */
+    public <T> CompletableFuture<ResponseEntity<T>> execute(Class<? extends T> clazz) {
+        requireNonNull(clazz, "clazz");
+        @SuppressWarnings("ThreadJoinLoop")
```

Comment:
It seems like cruft?

Reply:
I don't know why IntelliJ warn about this. But now I checked it again and the warning has gone. 
It seems an IntelliJ bug that was fixed by upgrading. 🤪